packages/gatsby-theme-newrelic/plugins/gatsby-source-swiftype/gatsby-node.js

```diff
@@ -0,0 +1,93 @@
+const fs = require('fs');
+const createRelatedResourceNode = require('./src/createRelatedResourceNode');
+const getRelatedResources = require('./src/getRelatedResources');
+
+const writeableData = {};
+
+exports.onPreBootstrap = (_, pluginOptions) => {
+  const { file } = pluginOptions;
+
+  if (!fs.existsSync(file)) {
+    fs.writeFileSync(file, '{}');
+  }
+};
+
+exports.onCreateNode = async (
+  { actions, node, getNodesByType, createNodeId, createContentDigest },
+  pluginOptions
+) => {
+  const { createNode, createParentChildLink } = actions;
+  const { filterNode = () => false, getPath } = pluginOptions;
+
+  if (node.internal.type !== 'Mdx' || !filterNode({ node })) {
+    return;
+  }
+
+  const [
+    {
+      siteMetadata: { siteUrl },
+    },
+  ] = getNodesByType('Site');
+
+  const pathname = getPath({ node });
+  const resources = await getRelatedResources({ node, siteUrl }, pluginOptions);
+
+  writeableData[pathname] = resources;
+
+  resources.forEach((resource) => {
+    const child = createRelatedResourceNode({
+      parent: node.id,
+      resource,
+      createContentDigest,
+      createNode,
+      createNodeId,
+    });
+
+    createParentChildLink({ parent: node, child: child });
```

Comment:
I'm guessing that in order to get the related resources we'll query for the children of the node?

Reply:
The PR is just copying what we had before. I didn't write the original code, but I believe we need to establish a relationship so that we can query related content for a specific page.

Reply:
Yes. This creates a `childrenRelatedPages` (if I remember correctly) field on the parent so that you can query for it for the current page.